src/Features/Core/Portable/Diagnostics/IRemoteDiagnosticAnalyzerService.cs

```diff
@@ -4,12 +4,13 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     interface IRemoteDiagnosticAnalyzerService
     {
-        Task CalculateDiagnosticsAsync(DiagnosticArguments arguments, string streamName, CancellationToken cancellationToken);
+        Task CalculateDiagnosticsAsync(PinnedSolutionInfo solutionInfo, DiagnosticArguments arguments, string streamName, CancellationToken cancellationToken);
```

Comment:
what's the intuition on what a PinnedSolutionInfo  is?

Reply:
I'll rename it to `SerializedSolutionInfo` later.